### PR TITLE
BIGTOP-4137: Add Solr component on Bigtop-3.3.0 stack 

### PR DESCRIPTION
**Pull request by lvkaihua**
Add Solr component on Bigtop-3.3.0 stack

### CODE REVIEW

**Comment by lvkaihua** [comment]
屏幕截图 2024-08-12 232359

File: bigtop-manager-server/src/main/resources/stacks/bigtop/3.3.0/services/solr/metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                <component>
+                    <name>SOLR</name>
+                    <display-name>Solr Instance</display-name>
+                    <category>MASTER</category>
+                    <cardinality>1+</cardinality>
+                    <command-script>
+                        <script-id>org.apache.bigtop.manager.stack.bigtop.v3_3_0.solr.SolrServerScript</script-id>
+                        <script-type>java</script-type>
+                        <timeout>1200</timeout>
+                    </command-script>
```
Comment (kevinw66) [inline bigtop-manager-server/src/main/resources/stacks/bigtop/3.3.0/services/solr/metainfo.xml:41]: Could you add quicklinks here? you can refers to [hdfs](https://github.com/apache/bigtop-manager/blob/main/bigtop-manager-server/src/main/resources/stacks/bigtop/3.3.0/services/hdfs/metainfo.xml#L58) config

File: bigtop-manager-stack/bigtop-manager-stack-bigtop/src/main/java/org/apache/bigtop/manager/stack/bigtop/v3_3_0/solr/SolrParams.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    public String get_znode(){
+        Map<String, Object> solrEnv = LocalSettings.configurations(serviceName(), "solr-env");
+        String znode = (String) solrEnv.get("solr_znode");
+        return znode;
+    }
+    public String get_zkstring(){
+        List<String> zookeeperServerHosts = LocalSettings.hosts("zookeeper_server");
+        Map<String, Object> zkport = LocalSettings.configurations("zookeeper", "zoo.cfg");
+        String clientPort = (String) zkport.get("clientPort");
+        String solrzkstring = MessageFormat.format("{0}:{1}", zookeeperServerHosts.get(0),clientPort);
```
Comment (kevinw66) [inline bigtop-manager-stack/bigtop-manager-stack-bigtop/src/main/java/org/apache/bigtop/manager/stack/bigtop/v3_3_0/solr/SolrParams.java:71]: Both method name and variable name should be camelcase

File: bigtop-manager-server/src/main/resources/stacks/bigtop/3.3.0/services/solr/configuration/solr-env.xml [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+  <property>
+    <name>security_enabled</name>
+    <value>false</value>
+  </property>
+
+  <!-- solr-env.sh -->
+  <property>
+    <name>content</name>
+    <display-name>solr-env template</display-name>
+    <description>This is the jinja template for solr-env.sh file</description>
```
Comment (kevinw66) [inline bigtop-manager-server/src/main/resources/stacks/bigtop/3.3.0/services/solr/configuration/solr-env.xml:224]: We are using freemarker template, not jinja

File: bigtop-manager-server/src/main/resources/stacks/bigtop/3.3.0/services/solr/configuration/solr-env.xml
Comment (kevinw66) [inline bigtop-manager-server/src/main/resources/stacks/bigtop/3.3.0/services/solr/configuration/solr-env.xml:1]: Xml file should also be indent with 4 spaces

File: bigtop-manager-server/src/main/resources/stacks/bigtop/3.3.0/services/solr/metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            <display-name>Solr</display-name>
+            <desc>
+                Solr is the popular, blazing-fast, open source enterprise search platform built on Apache Lucene.
+            </desc>
+            <version>8.11.2</version>
+            <user>solr</user>
+            <group>solr</group>
+            <components>
+                <component>
+                    <name>SOLR</name>
```
Comment (kevinw66) [inline bigtop-manager-server/src/main/resources/stacks/bigtop/3.3.0/services/solr/metainfo.xml:33]: Make sure component name is lowercase and similar to display name, it should be `solr_instance` here

File: bigtop-manager-server/src/main/resources/stacks/bigtop/3.3.0/services/solr/order.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+    "SOLR-START" : ["ZOOKEEPER_SERVER-START"]
```
Comment (kevinw66) [inline bigtop-manager-server/src/main/resources/stacks/bigtop/3.3.0/services/solr/order.json:2]: Also need to change to SOLR_INSTANCE-START

File: bigtop-manager-server/src/main/resources/stacks/bigtop/3.3.0/services/solr/metainfo.xml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+-->
+<metainfo>
+    <schema-version>2.0</schema-version>
+    <service>
+            <name>solr</name>
+            <display-name>Solr</display-name>
+            <desc>
+                Solr is the popular, blazing-fast, open source enterprise search platform built on Apache Lucene.
+            </desc>
+            <version>8.11.2</version>
```
Comment (kevinw66) [inline bigtop-manager-server/src/main/resources/stacks/bigtop/3.3.0/services/solr/metainfo.xml:28]: We should align to bigtop's version, which contains the release version, it should be 8.11.2-2 here

File: bigtop-manager-server/src/main/resources/stacks/bigtop/3.3.0/services/solr/metainfo.xml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+            <user>solr</user>
+            <group>solr</group>
+            <components>
+                <component>
+                    <name>SOLR</name>
+                    <display-name>Solr Instance</display-name>
+                    <category>MASTER</category>
```
Comment (kevinw66) [inline bigtop-manager-server/src/main/resources/stacks/bigtop/3.3.0/services/solr/metainfo.xml:35]: Also lowercase here

File: bigtop-manager-server/src/main/resources/stacks/bigtop/3.3.0/services/solr/configuration/solr-log4j.xml [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+  <property>
+    <name>log_maxbackupindex</name>
+    <value>9</value>
+    <description>The number of backup files</description>
+    <display-name>Solr Log: # of backup files</display-name>
+  </property>
+  <property>
+    <name>content</name>
+    <display-name>solr-log4j template</display-name>
+    <description>This is the jinja template for log4j2.xml</description>
```
Comment (kevinw66) [inline bigtop-manager-server/src/main/resources/stacks/bigtop/3.3.0/services/solr/configuration/solr-log4j.xml:37]: Also freemarker here

**Review by kevinw66 — CHANGES_REQUESTED** [review]

**Comment by kevinw66** [comment]
Could you also fix the code style?

**Comment by kevinw66** [comment]
There will be conflicts that need to be fixed after I merge #39

**Review by kevinw66 — APPROVED** [review]
+1 LGTM, thanks @lvkaihua

**Pull request merged**